config/recipes/maps/03-ingress.yaml

```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ems-demo
+  annotations:
+    # Disable HTTP traffic between the Ingress and the backend services
```

Comment:
According to [this documentation](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress-xlb#disabling_http):

> If you want all traffic between the client and the HTTP(S) load balancer to use HTTPS, you can disable HTTP by including the `kubernetes.io/ingress.allow-http` annotation in your Ingress manifest. Set the value of the annotation to "false".

IIUC `kubernetes.io/ingress.allow-http` affects the traffic between the client and the ingress point, not between the ingress and the backends.

Reply:
Oh well spotted. I think we might just drop the annotation all together I think it is GKE specific.

Reply:
If it helps here is what I'm using for my test (`networking.k8s.io/v1beta1` because I'm testing on K8S 1.18):

```yaml
---
apiVersion: networking.k8s.io/v1beta1
kind: Ingress
metadata:
  name: ems-demo
  annotations:
    cert-manager.io/issuer: "letsencrypt-prod"
    kubernetes.io/ingress.class: "nginx"
    nginx.ingress.kubernetes.io/backend-protocol: HTTPS
    nginx.ingress.kubernetes.io/proxy-ssl-verify: "off"
    nginx.ingress.kubernetes.io/ssl-redirect: "true"
spec:
  tls:
    - hosts: ["kibana.{{YOUR_DOMAIN}}"]
      # secret populated with your Kibana certificates
      secretName: kibana-certs
    - hosts: ["maps.{{YOUR_DOMAIN}}"]
      # secret populate with your Elastic Maps Server certificates
      secretName: maps-certs
  rules:
    - host: "maps.{{YOUR_DOMAIN}}"
      http:
        paths:
          - path: "/"
            #pathType: Prefix
            backend:
                    #  service:
              serviceName: quickstart-ems-http
              servicePort: 8080
    - host: "kibana.{{YOUR_DOMAIN}}"
      http:
        paths:
          - path: "/"
            #pathType: Prefix
            backend:
                    #  service:
              serviceName: quickstart-kb-http
              servicePort: 5601
```

I don't know how to avoid the `nginx.ingress.kubernetes.io/proxy-ssl-verify: "off"`

Reply:
I am using a similar Ingress. Do you think we should just commit to Nginx in the recipe and add the necessary annotations? 

> I don't know how to avoid the nginx.ingress.kubernetes.io/proxy-ssl-verify: "off"

I think in theory `nginx.ingress.kubernetes.io/proxy-ssl-secret: secretName` should work but we have different certs for the different backends.

Reply:
This is what I am using basically the same just a `v1` Ingress and I manually issued the let's encrypt certs

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: ems-in
  labels:
    app: ems-demo
  annotations:
    kubernetes.io/ingress.class: "nginx"
    nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
    nginx.ingress.kubernetes.io/proxy-ssl-verify: "off"
spec:
  tls:
    - hosts: ["dev.pebrc.ovh"]
      secretName: dev-pebrc-ovh
    - hosts: ["maps.pebrc.ovh"]
      secretName: maps-pebrc-ovh
  rules:
    - host: "maps.pebrc.ovh"
      http:
        paths:
          - path: "/"
            pathType: Prefix
            backend:
              service:
                name: ems-sample-ems-http
                port:
                  name: https
    - host: "dev.pebrc.ovh"
      http:
        paths:
          - path: "/"
            pathType: Prefix
            backend:
              service:
                name: thor-kb-http
                port:
                  name: https

```

Reply:
>  Do you think we should just commit to Nginx in the recipe and add the necessary annotations?

I'm not an ingress expert but it looks difficult to provide a working example compatible with all the ingress controllers. The Nginx implementation is developed as part of the K8S project and is available in Minikube, I think I would be inclined to provide a working example for Nginx and mention that it must be adapted for other ingress implementations.

> I think in theory `nginx.ingress.kubernetes.io/proxy-ssl-secret: secretName` should work but we have different certs for the different backends.

Yes that's the tricky part.